js_modules/dagster-ui/packages/ui-core/src/ui/formatDuration.ts

```diff
@@ -0,0 +1,156 @@
+interface DurationOptions {
+  /** Maximum value before the next unit is displayed */
+  maxValueBeforeNextUnit?: Record<UnitType, number>;
+  /** Number of significant digits to display (1 or 2) */
+  significantDigits?: 1 | 2;
+  /** Whether the input is in seconds (default: false, assumes milliseconds) */
+  unit?: 'seconds' | 'milliseconds';
+}
+
+type UnitType = 'year' | 'month' | 'week' | 'day' | 'hour' | 'minute' | 'second' | 'millisecond';
+type PluralUnitType = `${UnitType}s`;
+
+const defaultMaxValueBeforeNextUnit: Record<UnitType, number> = {
+  year: Infinity,
+  month: 24,
+  week: 8,
+  day: 14,
+  hour: 48,
+  minute: 120,
+  second: 120,
+  millisecond: 500,
+};
+
+interface DurationPart {
+  value: number;
+  unit: UnitType | PluralUnitType;
+}
+
+const UNITS: Array<[number, UnitType, PluralUnitType]> = [
+  [365 * 24 * 60 * 60 * 1000, 'year', 'years'],
+  [30 * 24 * 60 * 60 * 1000, 'month', 'months'],
+  [7 * 24 * 60 * 60 * 1000, 'week', 'weeks'],
+  [24 * 60 * 60 * 1000, 'day', 'days'],
+  [60 * 60 * 1000, 'hour', 'hours'],
+  [60 * 1000, 'minute', 'minutes'],
+  [1000, 'second', 'seconds'],
+  [1, 'millisecond', 'milliseconds'],
+];
+
+const YEAR_MS = 365 * 24 * 60 * 60 * 1000;
+const MONTH_MS = 30 * 24 * 60 * 60 * 1000;
+const WEEK_MS = 7 * 24 * 60 * 60 * 1000;
+const DAY_MS = 24 * 60 * 60 * 1000;
+
+/**
+ * Converts a duration in milliseconds or seconds to a human-readable format
+ * @param duration - The duration in milliseconds (default) or seconds
+ * @param options - Configuration options
+ * @returns Human-readable duration string
+ */
+export function formatDuration(duration: number, options: DurationOptions = {}): DurationPart[] {
+  const {
+    maxValueBeforeNextUnit = defaultMaxValueBeforeNextUnit,
+    significantDigits = 1,
+    unit = 'milliseconds',
+  } = options;
+
+  // Convert to milliseconds if input is in seconds
+  const ms = unit === 'seconds' ? duration * 1000 : duration;
```

Comment:
Seems unintuitive here that any `unit` passed in other than 'seconds' gets interpreted as milliseconds.

Reply:
check line 7, we only allow seconds or milliseconds since that is what the backend returns